knoten/csm.py

```diff
@@ -125,6 +121,13 @@ def generate_latlon_boundary(camera, boundary, semi_major=3396190, semi_minor=33
            List of altitude values
     '''
 
+    if not semi_major:
+        ellipse = csmapi.SettableEllipsoid.getEllipsoid(camera)
+        semi_major = ellipse.getSemiMajorRadius()
+    if not semi_minor:
+        ellipse = csmapi.SettableEllipsoid.getEllipsoid(camera)
+        semi_minor = ellipse.getSemiMinorRadius()
+
```

Comment:
I may need to back this change out until the SettableEllipsoid class is wrapped in swig.